### PR TITLE
Changelog-related changes

### PR DESCRIPTION
**Pull request by PLangowski**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,283 +2,520 @@
 
 All notable changes to this project will be documented in this file.
 
-The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
-and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
```
Comment (m-iwanicki) [inline CHANGELOG.md]: It doesn't as we use `-rc<x>` for release candidates. Semver uses `-rc.<x>`

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,283 +2,520 @@
...
 
-Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).
+## [0.7.5] - 2024-03-12
```
Comment (m-iwanicki) [inline CHANGELOG.md:85]: This and every other link generated by headers is in this format: https://github.com///compare/v0.7.4..v0.7.5
which isn't correct and results in 404 page not found

**Review by m-iwanicki — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (PLangowski) [reply CHANGELOG.md]: Done https://github.com/3mdeb/meta-rte/compare/59973da3165f4b6227a4a5eb9279c0cb3b63d074..c3d9963d89997d076211040aeac3e5fba9cab212#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edL6

File: CHANGELOG.md (hunk 2 above)
Comment (PLangowski) [reply CHANGELOG.md:85]: Fixed https://github.com/3mdeb/meta-rte/compare/59973da3165f4b6227a4a5eb9279c0cb3b63d074..c3d9963d89997d076211040aeac3e5fba9cab212#diff-e1372c8b03c40942b5d828a90975054cb8aaed3b38189f434396f922ec41a584R80

**Comment by PLangowski** [comment]
The config before trimmed the beginning of commit messages. Now the full short message is displayed. I also removed duplicates and some things that I thought didn't fit in the changelog. c3d9963d89997d076211040aeac3e5fba9cab212

**Comment by m-iwanicki** [comment]
@PLangowski If you force push then I think it's easier to see changes you made if you use compare commits, e.g. https://github.com/3mdeb/meta-rte/compare/59973da3165f4b6227a4a5eb9279c0cb3b63d074..c3d9963d89997d076211040aeac3e5fba9cab212.
It's hard to review new changes otherwise.

**Review by m-iwanicki — APPROVED** [review]
Looks ok to me, maybe remove "updated changelog" from changelog? Seems a little redundant.

**Comment by PLangowski** [comment]
@m-iwanicki I deleted the commits that add release notes to the changelog
https://github.com/3mdeb/meta-rte/compare/c3d9963d89997d076211040aeac3e5fba9cab212..ea6eccada1c864750669df863fa1034305b58781

**Review by m-iwanicki — CHANGES_REQUESTED** [review]
Fix pre-commit errors

File: scripts/generate-changed-recipes [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/python
```
Comment (m-iwanicki) [inline scripts/generate-changed-recipes]:
```suggestion
#!/usr/bin/env python
```

Add `.py` to script name

**Review by m-iwanicki — CHANGES_REQUESTED** [review]
Maybe move `generate-changelog.sh` to scripts dir?

File: scripts/generate-changed-recipes (hunk 3 above)
Comment (PLangowski) [reply scripts/generate-changed-recipes]: 0954429b463c5276be7898e733b00a87dac3b2ca

**Comment by PLangowski** [comment]
@m-iwanicki I think it's better if it remains in the root directory, so that it's visible. It's very important that contributors use this script instead of manually filling up the changelog, so we shouldn't reduce its visibility. Also, the script was in the root directory before this PR,

**Comment by m-iwanicki** [comment]
@PLangowski Only people doing releases will use this script and I hope they won't use it just because they saw it in root directory but because they know release process and are following it.

**Comment by m-iwanicki** [comment]
And what about
> Use the generate-changed-recipes script to print the list of changes in recipe versions [...]
> Paste the script output into your release's section in CHANGELOG.md [...]

in https://docs.zarhus.com/development-process/standard-release-process/#process-steps?

Maybe we should mention where to exactly paste the output or maybe add it in script after generating changelog step?

**Comment by PLangowski** [comment]
Maybe we should. You can create an issue for that. As for the `generte-changelog` script, I moved it b012297bde02720426c8b18b87f12e95d3d26021

**Review by m-iwanicki — APPROVED** [review]

**Pull request merged**